scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2110,6 +2126,12 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     if (token.is[LF] && ahead { token.is[T] }) newLineOpt()
   }
 
+  def newLineOptWhenFollowedByIndentationAnd[T <: Token: TokenInfo]: Unit = {
+    val prev = currentIndentation
```

Comment:
this operation is pretty expensive. I'd check for LF and dialect flag first, then do this in the block.

Reply:
sure, moved it to a separate def together with the ahead check.